Lib/dbm/__init__.py

```diff
@@ -109,6 +110,7 @@ def whichdb(filename):
     """
 
     # Check for ndbm first -- this has a .pag and a .dir file
+    filename = os.fspath(filename)
```

Comment:
I some cases it is better to use `os.fsdecode()`. It will add support of bytes paths.

Reply:
`os.fsencode()`, not `os.fsdecode()`, isn't it?  We have to take the object whatever it is (`str`, `bytes` or any object implementing `__fspath__`) and convert that to a `bytes` object.

Note that after `os.fsencode`, all extensions concatenated to that path should be bytes.